packages/timeline-state-resolver/src/integrations/websocketTcpClient/__tests__/websocketTcpClient.spec.ts

```diff
@@ -0,0 +1,229 @@
+import { jest } from '@jest/globals'
+import {} from 'timeline-state-resolver-types/dist/integrations/websocketTcpClient'
+import { WebSocketTcpConnection } from '../connection'
+import { WebSocketTcpClientDevice, WebSocketTcpCommand } from '../index'
+import {
+	Timeline,
+	DeviceType,
+	WebSocketTCPClientOptions,
+	TimelineContentTypeWebSocketTcpClient,
+	StatusCode,
+	TSRTimelineContent,
+} from 'timeline-state-resolver-types'
+import { MockTime } from '../../../__tests__/mockTime'
+import { TimelineContentWebSocketTcpClientAny } from 'timeline-state-resolver-types/src'
+
+// Mock the WebSocketTcpConnection??
+jest.mock('../connection')
+
+const MockWebSocketTcpConnection = WebSocketTcpConnection as jest.MockedClass<typeof WebSocketTcpConnection>
+
+describe('WebSocketTcpClientDevice', () => {
+	const mockTime = new MockTime()
+	let device: WebSocketTcpClientDevice
+
+	beforeEach(async () => {
+		jest.clearAllMocks()
+		mockTime.init()
+
+		// Create device context
+		const deviceContext = {
+			getCurrentTime: mockTime.getCurrentTime,
+			getTimeSinceStart: 0,
+			logger: {
+				info: jest.fn(),
+				warn: jest.fn(),
+				error: jest.fn(),
+				debug: jest.fn(),
+			},
+			emit: jest.fn(),
+			resetStateCheck: jest.fn(),
+			timeTrace: jest.fn(),
+			commandError: jest.fn(),
+			resetToState: jest.fn(),
+		}
+
+		// Create mock options
+		const options: WebSocketTCPClientOptions = {
+			webSocket: {
+				uri: 'ws://localhost:8080',
+				reconnectInterval: 5000,
+			},
+			tcp: {
+				host: '127.0.0.1',
+				port: 1234,
+				bufferEncoding: 'utf8',
+			},
+		}
+
+		device = new WebSocketTcpClientDevice(deviceContext as any)
+
+		// Mock connection methods
+		MockWebSocketTcpConnection.prototype.connect.mockResolvedValue()
+		MockWebSocketTcpConnection.prototype.disconnect.mockResolvedValue()
+		MockWebSocketTcpConnection.prototype.connected.mockReturnValue(true)
+		MockWebSocketTcpConnection.prototype.sendWebSocketMessage.mockImplementation(() => {})
+		MockWebSocketTcpConnection.prototype.sendTcpMessage.mockImplementation(() => {})
+
+		// Initialize device
+		await device.init( options )
+	})
+
+	afterEach(() => {
+		//Are there something like??:
+		//mockTime.dispose()
+		// Or can we just ignore this
+	})
+
+	describe('Connections', () => {
+		test('init', async () => {
+			expect(MockWebSocketTcpConnection.prototype.connect).toHaveBeenCalled()
+		})
+
+		test('terminate', async () => {
+			await device.terminate()
```

Comment:
This could cause race conditions when this test is ran in parallel with other tests